google-cloud-pub-sub/src/main/scala/akka/stream/alpakka/googlecloud/pubsub/scaladsl/GooglePubSub.scala

```diff
@@ -23,65 +23,95 @@ protected[pubsub] trait GooglePubSub {
   private[pubsub] def httpApi: PubSubApi
 
   /**
-   * Creates a flow to that publish messages to a topic and emits the message ids
+   * Creates a flow to that publishes messages to a topic and emits the message ids.
    */
-  def publish(topic: String, config: PubSubConfig, parallelism: Int = 1)(
-      implicit actorSystem: ActorSystem,
-      materializer: Materializer
-  ): Flow[PublishRequest, immutable.Seq[String], NotUsed] =
+  def publish(topic: String,
+              config: PubSubConfig,
+              parallelism: Int = 1): Flow[PublishRequest, immutable.Seq[String], NotUsed] =
     Flow[PublishRequest]
       .map((_, ()))
       .via(
         publishWithContext[Unit](topic, config, parallelism).asFlow
       )
       .map(_._1)
 
-  def publishWithContext[C](topic: String, config: PubSubConfig, parallelism: Int = 1)(
-      implicit actorSystem: ActorSystem,
-      materializer: Materializer
-  ): FlowWithContext[PublishRequest, C, immutable.Seq[String], C, NotUsed] =
-    httpApi
-      .accessTokenWithContext[PublishRequest, C](config)
-      .via(
-        httpApi.publish[C](config.projectId, topic, parallelism)
-      )
+  /**
+   * Creates a flow to that publishes messages to a topic and emits the message ids and carries a context
+   * through.
+   */
+  def publishWithContext[C](
+      topic: String,
+      config: PubSubConfig,
+      parallelism: Int = 1
+  ): FlowWithContext[PublishRequest, C, immutable.Seq[String], C, NotUsed] = {
+    // some wrapping back and forth as FlowWithContext doesn't offer `setup`
+    // https://github.com/akka/akka/issues/27883
+    FlowWithContext.fromTuples {
+      Flow
+        .setup { (mat, _) =>
```

Comment:
`Flow.setup` is deprecated in Akka 2.6, but the recommended alternative doesn't exist in 2.5 (`Flow.fromMaterializer`).  Maybe we can add a comment to change it when 2.5 is dropped.

Reply:
We'll revisit the Akka deprecations when we drop Akka 2.5 (in a year or so).